### ISSUE TITLE
Vega-expression support for string template literals

### ISSUE THREAD

**Issue by danmarshall** [vega/vega#2347]
Hello, as I was outputting some text I thought that [string template literals](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals) might come in handy in the [expression language](https://vega.github.io/vega/docs/expressions/).

Currently, to achieve this data-driven text: 
```
The value of row 1 is "a"
```
The expression has much use of the `+` operator and single quotes:
```
'The value of row ' + datum.row_number + ' is \"' + datum.v + '\"'
```
With template literals it becomes:
``` 
`The value of row ${datum.row_number} is \"${datum.v}\"`
```

- [x] Check for duplicate issues. Please file separate requests as separate issues on GitHub.
- [x] Clearly describe the goal of the feature request. Why is it needed?
- [x] Describe an envisioned solution. What do you want the new behavior to be?
- [x] If applicable, share mockup images or links to examples illustrating the desired output.
- [x] If applicable, share JSON spec designs for how the behavior would be made available in Vega.

**Comment by domoritz** [comment]
This would mean that any browser has to have support for string templates unless we compile down to concatenation.

**Comment by danmarshall** [comment]
True. Is it also true that expressions are "compiled" regardless?

**Comment by domoritz** [comment]
We always parse expressions and generate code. But the code mostly stays the same.

**Comment by hydrosquall** [comment]
Is there a minimum browser version that Vega commits to supporting for new versions?

I recognize that older browsers can be used for many years, just noting that browser support for this (in non CSP mode) is widespread per MDN as of 2020, though admittedly this is not everybody https://caniuse.com/template-literals

The tricky part here may be updating vega-expression to support the syntax, in particular our big parser.js file

**Comment by danmarshall** [comment]
Can a polyfill be used? Something like https://github.com/babel/babel/tree/main/packages/babel-plugin-transform-template-literals

**Comment by hydrosquall** [comment]
I’m not sure if a polyfill will help for parsing since we’re not already
using Babel, but that’s an interesting idea if we decided to transpile the
expressions before running our parser, thank you. The file we’d have to
modify is this copy (or fork?) of esprima.

https://github.com/vega/vega/blob/main/packages/vega-expression/src/parser.js
